internal/service/alarms/internal/db/repo/alarms_repository.go

```diff
@@ -137,3 +146,150 @@ func (ar *AlarmsRepository) UpsertAlarmDefinitions(ctx context.Context, records
 
 	return utils.ExecuteCollectRows[models.AlarmDefinition](ctx, ar.Db, sql, args)
 }
+
+// UpsertAlarmEventRecord insert and updating an AlarmEventRecord.
+func (ar *AlarmsRepository) UpsertAlarmEventRecord(ctx context.Context, records []models.AlarmEventRecord) error {
+	// Build queries for each record
+	sql, params, err := buildAlarmEventRecordUpsertQuery(records)
+	if err != nil {
+		return fmt.Errorf("failed to build query for event upsert: %w", err)
+	}
+
+	_, err = ar.Db.Exec(ctx, sql, params...)
+	if err != nil {
+		return fmt.Errorf("failed to execute upsert query: %w", err)
+	}
+
+	slog.Info("Successfully inserted alerts from alertmanager", "table", models.AlarmEventRecord{}.TableName(), "count", len(records))
+	return nil
+}
+
+// buildAlarmEventRecordUpsertQuery builds the query for insert and updating an AlarmEventRecord
+func buildAlarmEventRecordUpsertQuery(records []models.AlarmEventRecord) (string, []any, error) {
+	query := psql.Insert(im.Into(models.AlarmEventRecord{}.TableName()))
+
+	// Set cols
+	query.Expression.Columns = utils.GetColumns(records[0], []string{
+		"AlarmRaisedTime", "AlarmChangedTime", "AlarmClearedTime",
+		"AlarmAcknowledgedTime", "AlarmAcknowledged", "PerceivedSeverity",
+		"Extensions", "ObjectID", "ObjectTypeID",
+		"AlarmStatus", "Fingerprint", "AlarmDefinitionID",
+		"ProbableCauseID",
+	})
+
+	// Set values
+	values := make([]bob.Mod[*dialect.InsertQuery], 0, len(records))
+	for _, record := range records {
+		values = append(values, im.Values(psql.Arg(
+			record.AlarmRaisedTime, record.AlarmChangedTime, record.AlarmClearedTime,
+			record.AlarmAcknowledgedTime, record.AlarmAcknowledged, record.PerceivedSeverity,
+			record.Extensions, record.ObjectID, record.ObjectTypeID,
+			record.AlarmStatus, record.Fingerprint, record.AlarmDefinitionID,
+			record.ProbableCauseID,
+		)))
+	}
+	query.Apply(values...)
+
+	// Set upsert constraints
+	query.Apply(buildUpsertConstraints())
+
+	return query.Build() //nolint:wrapcheck
+}
+
+// buildUpsertConstraints creates the conflict resolution modifiers
+func buildUpsertConstraints() bob.Mod[*dialect.InsertQuery] {
+	dbTags := utils.GetAllDBTagsFromStruct(models.AlarmDefinition{})
+	return im.OnConflictOnConstraint("unique_fingerprint_alarm_raised_time").DoUpdate(
```

Comment:
the db/model has a OnConflict method for this

Reply:
updated